### PR TITLE
SEG-3379: Segment Query Builder Polish 🇵🇱

### PR DESCRIPTION
**Pull request by reedsa**
### [SEG-3379](https://jira.sendgrid.net/browse/SEG-3379)

Improvements to the query builder component
* Fixes console error from extraneous prop
* Fixes transitions and alignment of multiple conditions
* Configurable clear icon on the date picker

## Merge Checklist

- [X] Includes link to associated Github Issue or JIRA ticket
- [X] Has descriptive PR title and body
- [X] All changes are backwards compatible
- [ ] Code reviewed by a SendGrid frontend lead

### CODE REVIEW

File: src/datepicker.tsx [hunk 1]
```diff
@@ -34,7 +34,6 @@ export class SingleDatePickerWithRange extends Component<
           onFocusChange={onFocusChange}
           renderMonthElement={monthProps => this.renderMonthElement(monthProps)}
           daySize={50}
-          showClearDate
```
Comment (Ditofry) [inline src/datepicker.tsx]: I'm happy to just get rid of this, but are you sure that other parts of the UI aren't counting on it?  ARE there other parts of the UI besides custom fields and the QB that pertain to this even?

Comment (reedsa) [reply src/datepicker.tsx]: Yeah I made the mistake of having this originally, but AFAIK I was the only one using this particular component. I updated MCUI to reflect this change in the one component that is using it as well.

File: src/styles/segment-term.module.scss [hunk 2]
```diff
@@ -189,6 +198,13 @@
     cursor: pointer;
     opacity: 0;
   }
+
+  > .segment-term-text {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Comment (Ditofry) [inline src/styles/segment-term.module.scss]: 👍

File: src/datepicker.tsx (hunk 1 above)
Comment (Ditofry) [reply src/datepicker.tsx]: nvmnvmnvm

Comment (Ditofry) [reply src/datepicker.tsx]: I wasn't thinking about the fact that the prop is now passthrough to the airbnb component underneath

**Comment by thinkingserious** [comment]
Hello @reedsa,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**